sudospawner/spawner.py

```diff
@@ -94,6 +94,19 @@ def start(self):
 
     @gen.coroutine
     def _signal(self, sig):
+        if sig == 0:
+            # short-circuit existence check without invoking sudo
+            try:
+                os.kill(self.pid, sig)
+            except ProcessLookupError:
+                # No such process
+                return False
+            except PermissionError:
+                # When running hub with reduced permissions,
+                # we won't have permission to send signals, even 0.
+                # PermissionError means process exists.
```

Comment:
I agree this should be a rare occurrence. As I think about it more, the problem likely exists already when the check runs with sudo. The PID will exist, but it's not guaranteed to be the notebook server.

Thanks for the think-through.